internal/common/objects/BaseBiker.go

```diff
@@ -181,6 +197,40 @@ func (bb *BaseBiker) GetResourceAllocationParams() ResourceAllocationParams {
 	return bb.allocationParams
 }
 
+// default implementation returns the id of the nearest lootbox
+func (bb *BaseBiker) ProposeDirection() uuid.UUID {
+	return bb.NearestLoot()
+}
+
+func (bb *BaseBiker) ToggleOnBike() {
+	bb.onBike = !bb.onBike
+}
```

Comment:
Should this be exported? Given you're passing a pointer receiver, any agent can call this function to set their external state. 

This should be in /controlled by the server